httpserver/http_connection.cpp

```diff
@@ -155,12 +158,31 @@ accept_connection(boost::asio::io_context& ioc,
                 ->start();
 
         if (ec) {
-            LOKI_LOG(error, "Could not accept a new connection {}: {}",
-                     ec.value(), ec.message());
+            LOKI_LOG(
+                error,
+                "Could not accept a new connection {}: {}. Will only start "
+                "accepting new connections after a short delay.",
+                ec.value(), ec.message());
+
+            // If we fail here we are unlikely to be able to accept a new
+            // connection immediately, hence the delay
+            acceptor_timer.expires_after(ACCEPT_DELAY);
+            acceptor_timer.async_wait([&](const error_code& ec) {
+                if (ec && ec != boost::asio::error::operation_aborted) {
+                    // Not sure how to recover here, so it is probably the
+                    // safest to simply abort and let the launcher/systemd
+                    // restart us
+                    abort();
+                }
+
+                accept_connection(ioc, ssl_ctx, acceptor, sn,
+                                  channel_encryption, rate_limiter, security);
+            });
+        } else {
+            accept_connection(ioc, ssl_ctx, acceptor, sn, channel_encryption,
+                              rate_limiter, security);
```

Comment:
Slightly cleaner: you can put this in the `if (!ec)` above and then make the entire `if (ec) { ... }` just an `else` of that `if` statement.  (Currently it looks a bit odd to do: `if (a) ...; if (!a) ... else ...;`)

Reply:
True, updated.